src/runtime/components/sanity-content.ts

```diff
@@ -107,9 +108,20 @@ function findSerializer (item: Block | undefined, serializers: Required<Serializ
   return item?._type ? serializers.types[item._type] || serializers.marks[item._type] : undefined
 }
 
-function renderStyle ({ style, listItem }: Block, serializers: Required<Serializers>, children?: () => Children) {
-  if (!listItem && style && serializers.styles[style]) {
-    return h(serializers.styles[style] as any, null, isVue2 ? children?.() : { default: children })
+function renderStyle (item: Block, serializers: Required<Serializers>, children?: () => Children) {
+  const { style, listItem } = item
+  const serializer = serializers.styles[style]
+  const isElement = typeof serializer === 'string'
+  const props = Object.fromEntries(Object.entries(item).filter(([key]) => key !== '_type' && key !== 'markDefs' && key !== 'children').map(([key, value]) => {
```

Comment:
what was the issue with `children`?

Reply:
> what was the issue with `children`?

@danielroe Received a rendering warning, admittedly I didn't dig too deeply

```
[Vue warn]: Failed setting prop "children" on <p>: value [object Object] is invalid. TypeError: Cannot set property children of #<Element> which has only a getter
    at patchDOMProp (runtime-dom.esm-bundler.js:277:12)
    at patchProp (runtime-dom.esm-bundler.js:410:9)
    at mountElement (runtime-core.esm-bundler.js:5174:25)
    at processElement (runtime-core.esm-bundler.js:5137:13)
    at patch (runtime-core.esm-bundler.js:5057:21)
    at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:5609:21)
    at ReactiveEffect.run (reactivity.esm-bundler.js:185:25)
    at instance.update (runtime-core.esm-bundler.js:5716:56)
    at setupRenderEffect (runtime-core.esm-bundler.js:5730:9)
    at mountComponent (runtime-core.esm-bundler.js:5512:9) 
  at <Normal key="a8238f1a854b" children= [{…}] style= {} > 
  at <AsyncComponentWrapper key="a8238f1a854b" children= [{…}] style="normal" > 
  at <SanityContent blocks= (3) [{…}, {…}, {…}] serializers= {marks: {…}, types: {…}, styles: {…}} > 
  at <Section class="site-footer theme-dark" > 
  at <Footer key="footer" > 
  at <TransitionGroup key=1 > 
  at <[...slug] onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< Proxy {__v_skip: true} > > 
  at <Anonymous key="/" routeProps= {Component: {…}, route: {…}} pageKey="/" > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="page" mode="out-in" > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage> 
  at <App key=1 > 
  at <NuxtRoot>
```

Reply:
do you have an example of the block array that caused that?

Reply:
Example below.

I dug a bit more and the warning only happens when using a custom element to render the style and there's no defined prop `children`. Adding a prop for `children` removes the error. Considering this I'd just recommend adding the prop rather than disabling passing the children object as I've done. 

```[
    {
        "_key": "e5833f6c5d88",
        "_type": "block",
        "children": [
            {
                "_key": "0ea0635ddc510",
                "_type": "span",
                "marks": [],
                "text": "What we do"
            }
        ],
        "element": "h3",
        "markDefs": [],
        "style": "heading",
        "delay": 0.5
    },
]```